meps/mep-0004.md

```diff
@@ -0,0 +1,273 @@
+
+## Context
+
+### Current Limitations
+
+Code-cells from Jupyter Notebooks can produce rich output like Markdown, LaTeX, and tables. Right now, each output is effectively treated as a <wiki:Black_Box>, which is only interpreted at export time (to PDF, web, etc.) such that static exports and web build are required to interpret the MIME bundle outputs. As a result, the output content does not participate in a MyST build i.e. the Markdown or LaTeX output cannot generate or consume referencing labels.
+
+This limitation prevents programmatic generation of content that integrates with the rest of the document. For example, if a code cell generates Markdown output containing a figure with a label, that label cannot be referenced from other parts of the document, and the figure cannot appear in cross-reference resolution.
+
+### Use Cases
+
+This enhancement enables several important use cases:
+
+1. **Programmatic Content Generation**: Users can generate MyST Markdown from code cells, and this content can participate fully in the build process, including reference resolution.
+
+2. **Integrated Outputs**: Generated Markdown can define and consume reference targets that are visible to the rest of the project, enabling richer integration between computational content and documentation.
+
+3. **Executable Plugins Pattern**: Similar to executable plugins documentation for directives that can return MyST AST by printing to stdout, users should be able to do the same via `{code-cell}` directives and Jupyter Notebook cells. This would allow users to more flexibly generate MyST AST without requiring any kernel-specific libraries (since you'd just be using stdout).
+
+This proposal addresses the requirements described in [#1026](https://github.com/jupyter-book/mystmd/issues/1026), which tracks the need to associate AST subtrees with individual cell outputs.
+
+## Proposal
+
+### AST Structure Changes
+
+The MyST AST will be extended to support per-output AST representation through the following changes:
+
+**Previous Structure (Version 2):**
+
+```typescript
+type OutputV2 = {
+  type: 'output';
+  data?: any[]; // Array of IOutput bundles
+  visibility?: any;
+};
+```
+
+In this structure, a single `Output` node contained an **array** of output data bundles in the `data` field.
+
+**New Structure (Version 3):**
+
+```typescript
+type Output = {
```

Comment:
```suggestion
// Outputs (below) contains one or more Output nodes
type Output = {
```

Reply:
it confused me to see both of them at first, I thought you meant it's called 'Output' when there's one node, and 'Outputs' when there are more than one. But I think instead you're saying they're nested so I just tried to clarify